### PR TITLE
Wip node expansion

### PR DESCRIPTION
**Pull request by shreekarSS**

### CODE REVIEW

File: ocs_ci/ocs/constants.py [hunk 1]
```diff
@@ -68,6 +68,7 @@
 ROUTE = "Route"
 NODE = "Node"
 DEPLOYMENTCONFIG = "deploymentconfig"
+MACHINESETS = 'machinesets'
 # Provisioners
 AWS_EFS_PROVISIONER = "openshift.org/aws-efs"
```
Comment (hnallurv) [inline ocs_ci/ocs/constants.py]: where are you planning to use this?

File: tests/manage/cluster/cluster_expansion/test_node_expansion.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        ):
+            if len(sample) == count * len(machines):
+                break
+
+        worker_nodes_after = helpers.get_worker_nodes()
+        logger.info(f'after count of worker nodes {len(worker_nodes_after)}')
+        wait_for_nodes_status(
+            node_names=worker_nodes_after, status=constants.NODE_READY
+        )
+        # ToDo run IOs
```
Comment (hnallurv) [inline tests/manage/cluster/cluster_expansion/test_node_expansion.py]: Please refer to cluster expansion tests for expected behaviours for node addition and implement the same in the scripts

File: ocs_ci/ocs/machine.py [hunk 3]
```diff
@@ -120,3 +120,28 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
...
+
+
+def get_machinesets():
+    machines = list()
+    machinesets_obj = OCP(kind=constants.MACHINESETS, namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE)
+    for machine in machinesets_obj.get()['items']:
+        machines.append(machine.get('spec').get('selector').get(
+            'matchLabels').get('machine.openshift.io/cluster-api-machineset'))
+
+    return machines
```
Comment (hnallurv) [inline ocs_ci/ocs/machine.py]: from function name, it looks like it returns machinests but the return variable name is 'machines'. make it consistent

File: ocs_ci/ocs/machine.py [hunk 4]
```diff
@@ -120,3 +120,28 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
...
         return state == constants.STATUS_RUNNING.islower()
     return False
+
+
+def get_machinesets():
+    machines = list()
+    machinesets_obj = OCP(kind=constants.MACHINESETS, namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE)
+    for machine in machinesets_obj.get()['items']:
+        machines.append(machine.get('spec').get('selector').get(
+            'matchLabels').get('machine.openshift.io/cluster-api-machineset'))
```
Comment (hnallurv) [inline ocs_ci/ocs/machine.py]: can we have this defined in constants.py?

File: ocs_ci/ocs/machine.py [hunk 5]
```diff
@@ -120,3 +120,28 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
         log.info(f"{new_machine.name} is in {state} state")
         return state == constants.STATUS_RUNNING.islower()
     return False
+
+
+def get_machinesets():
```
Comment (hnallurv) [inline ocs_ci/ocs/machine.py]: Make sure there are functions to cover the node additions in VMware environment also.

File: ocs_ci/ocs/machine.py [hunk 6]
```diff
@@ -120,3 +120,28 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
...
+
+    return machines
+
+
+def get_replica_count(machine):
+    machinesets_obj = OCP(kind=constants.MACHINESETS, namespace=constants.OPENSHIFT_MACHINE_API_NAMESPACE)
+    return machinesets_obj.get(resource_name=machine).get('spec').get('replicas')
+
+
+def add_node(machine, count):
```
Comment (hnallurv) [inline ocs_ci/ocs/machine.py]: better to change the parameter name machine to machine_set

File: ocs_ci/ocs/machine.py [hunk 7]
```diff
@@ -120,3 +120,28 @@ def delete_machine_and_check_state_of_new_spinned_machine(machine_name):
...
+    run_cmd(f'oc -n {constants.OPENSHIFT_MACHINE_API_NAMESPACE} scale --replicas={count} machinesets {machine}')
+    return True
+
+
+def remove_node(machine, count):
```
Comment (hnallurv) [inline ocs_ci/ocs/machine.py]: Better to change the parameter name machine to machine_set

File: ocs_ci/ocs/constants.py [hunk 8]
```diff
@@ -290,11 +286,3 @@
     'https://quay.io/api/v1/repository/rhceph-dev/ocs-registry/'
     'tag/?onlyActiveTags=true&limit=2'
 )
-
-# Platforms
```
Comment (ebenahar) [inline ocs_ci/ocs/constants.py]: all these constants are in use. Any reason to delete them?

Comment (shreekarSS) [reply ocs_ci/ocs/constants.py]: My bad, it was old repo, corrected in pr785

**Pull request closed without merging**